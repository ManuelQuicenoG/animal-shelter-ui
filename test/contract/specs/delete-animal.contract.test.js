import { provider } from '../config/initPact';
import { AnimalController } from '../../../controllers';
import { Matchers } from '@pact-foundation/pact';

const animal_name = 'prueba';

describe('Given an animal service', () => {
    beforeAll(async() => {
        await provider.setup();
    });

    describe('When a request to remove a cat arrived ', () => {
        beforeAll(async() => {
            await provider.addInteraction({
                state: 'backend service is ready',
                uponReceiving: 'a request to delete a cat',
                withRequest: {
                    method: 'DELETE',
                    path: '/animals/'+animal_name
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.eachLike({
                        message: Matchers.string('Animal deleted successfully'),
                    }, {min: 1})
                }
            });
        });

        test('Then it should return a confirm message that indicate successfully process', async() => {
            const response = await AnimalController.delete(animal_name);
            expect(response.data).toMatchSnapshot();

            await provider.verify();
        })

    });

    afterAll(async () => {
        await provider.finalize();
    });
});
