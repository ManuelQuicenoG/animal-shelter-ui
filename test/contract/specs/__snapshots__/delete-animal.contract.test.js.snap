// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an animal service When a request to remove a cat arrived  Then it should return a confirm message that indicate successfully process 1`] = `
Array [
  Object {
    "message": "Animal deleted successfully",
  },
]
`;
